metrics_sdk/lib/opentelemetry/sdk/metrics/meter.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    # The Metrics module contains the OpenTelemetry metrics reference
+    # implementation.
+    module Metrics
+      # {Meter} is the SDK implementation of {OpenTelemetry::Metrics::Meter}.
+      class Meter < OpenTelemetry::Metrics::Meter
+        # @api private
+        #
+        # Returns a new {Meter} instance.
+        #
+        # @param [String] name Instrumentation package name
+        # @param [String] version Instrumentation package version
+        #
+        # @return [Meter]
+        def initialize(name, version, meter_provider)
+          @mutex = Mutex.new
+          @instrument_registry = {}
+          @instrumentation_library = InstrumentationLibrary.new(name, version)
+          @meter_provider = meter_provider
+        end
+
+        # @api private
+        def add_metric_reader(metric_reader)
+          @instrument_registry.each do |_n, instrument|
+            instrument.register_with_new_metric_store(metric_reader.metric_store)
+          end
+        end
+
+        def create_instrument(kind, name, unit, description, callback)
+          super do
+            case kind
```

Comment:
The intent is that `super do` will yield the return of the `case` statement to the superclass?